### PR TITLE
stats team

### PR DESCRIPTION
**Pull request by manland**
https://github.com/manland/mattermost-plugin-analytics/issues/8

Add team name before channel to know where channel is located

### CODE REVIEW

File: server/post.go [hunk 1]
```diff
@@ -64,15 +64,18 @@ func (p *Plugin) buildAnalyticMsg() (string, *model.AppError) {
...
 	}
 	urlPie.RawQuery = parametersURLPie.Encode()
-	pie := "![](" + urlPie.String() + ") "
+	pie := fmt.Sprintf("![channels pie chart](%s) ", urlPie.String())
 
 	urlBar, _ := url.Parse("http://127.0.0.1:8065/plugins/com.github.manland.mattermost-plugin-analytics/bar.svg")
 	parametersURLBar := url.Values{}
-	for _, c := range data.users {
+	for index, c := range data.users {
+		if index > 10 {
```
Comment (hanzei) [inline server/post.go]: why?

File: server/plugin.go [hunk 2]
```diff
@@ -142,25 +144,33 @@ func (p *Plugin) updateOrAppend(originals []analyticsData, upsert analyticsData)
 	return append(originals, upsert)
 }
 
-func (p *Plugin) getChannelName(key string) (string, string, *model.AppError) {
+func (p *Plugin) getChannelName(key string) (string, string, string, *model.AppError) {
```
Comment (hanzei) [inline server/plugin.go]: Would you please add a GoDoc comment to this method? It's not clear what the three return parameters are.

File: server/plugin.go [hunk 3]
```diff
@@ -142,25 +144,33 @@ func (p *Plugin) updateOrAppend(originals []analyticsData, upsert analyticsData)
...
 	if err != nil {
-		return "", "", &model.AppError{
+		return "", "", "", &model.AppError{
 			Message:       "Can't retreive channel name",
 			DetailedError: err.Error(),
 		}
 	}
 	if channel.IsGroupOrDirect() {
-		return "Private", "Private", nil
+		return "", "Private", "", nil
```
Comment (hanzei) [inline server/plugin.go]: I'm 3/5 not calling this `Private` as this is reserved to private channels. Maybe just call it `DM`?

File: server/post.go [hunk 4]
```diff
@@ -137,3 +140,10 @@ func (p *Plugin) sendAnalytics() *model.AppError {
 
 	return nil
 }
+
+func channelLink(displayName string, link string) string {
```
Comment (hanzei) [inline server/post.go]: Maybe just pass `analyticsData` instead of two strings?

Comment (hanzei) [inline server/post.go]: I would call it `getGhannelLink`.

File: server/post.go [hunk 5]
```diff
@@ -48,13 +48,13 @@ func (p *Plugin) buildAnalyticMsg() (string, *model.AppError) {
 
 		m = m + "### :see_no_evil: Podium Channels Conversations\n"
 		if len(data.channels) > 0 {
-			m = m + fmt.Sprintf("* :1st_place_medal: ~%s with a total of **%d** *(%d%%)* messages with %d reply\n", data.channels[0].name, data.channels[0].nb, (data.channels[0].nb*100)/(data.totalMessagesPublic+data.totalMessagesPrivate), data.channels[0].reply)
+			m = m + fmt.Sprintf("* :1st_place_medal: %s with a total of **%d** *(%d%%)* messages with %d reply\n", channelLink(data.channels[0].displayName, data.channels[0].link), data.channels[0].nb, (data.channels[0].nb*100)/(data.totalMessagesPublic+data.totalMessagesPrivate), data.channels[0].reply)
```
Comment (hanzei) [inline server/post.go]: These lines are way to complicate. Please split this up some way.

File: server/plugin.go [hunk 6]
```diff
@@ -51,8 +51,10 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*mo
 }
 
 type analyticsData struct {
```
Comment (hanzei) [inline server/plugin.go:64]: Mind adding a GoDoc comment for this struct as it's am important struct, but not straight forward to see what it does.

File: server/post.go (hunk 1 above)
Comment (manland) [reply server/post.go]: it's the number max to display a good bar chart. If we have 200 users, the chart will be unreadable. I have added a constant to be more comprehensive.

File: server/post.go (hunk 4 above)
Comment (manland) [reply server/post.go]: :heavy_check_mark:

Comment (manland) [reply server/post.go]: :heavy_check_mark:

File: server/plugin.go (hunk 3 above)
Comment (manland) [reply server/plugin.go]: :heavy_check_mark:

File: server/plugin.go (hunk 2 above)
Comment (manland) [reply server/plugin.go]: :heavy_check_mark:

File: server/plugin.go (hunk 6 above)
Comment (manland) [reply server/plugin.go:64]: :heavy_check_mark:

File: server/post.go (hunk 5 above)
Comment (manland) [reply server/post.go]: :heavy_check_mark:

**Pull request merged**